### PR TITLE
fix: Handled edge case for re-queued jobs handling sync output cli & some integ test tweaks

### PR DESCRIPTION
**Pull request by sakshie95**
### What was the problem/requirement? (What/Why)
1. We found an unhandled edge case with the integ tests for the sync-output a.k.a incremental download feature. In case of re-queued tasks sometimes the CLI ends up downloading the same output more than once due to the eventual consistency window if the tasks get executed and re-queued/re-executed within the same 2 min window. 
2. We found an issue with the test ``test_incremental_download_dep_data_flow`` where it looks in the wrong location for the checkpoint file and doesn't find it which results in output from the last task executed not being downloaded
3. Sometimes the integ tests also run out of memory

### What was the solution? (How)
1. Fixed the edge case by adding the eventual consistency window to ``session_ended_threshold`` only for updated and completed jobs and not new jobs. New jobs get the eventual consistency window in accordance with the bootstrap lookback in minutes which will be None in case of re-queued jobs not intended to be downloaded again.
2. Fixed the location of the the checkpoint file in the test
3. Tried to limit the continously running loop until job completes to 10 minutes given that the test times out in 15 minutes. This might fix the memory leak but we'll keep monitoring. 

We also added the option ``--xfail-tb`` for traceback for xfail tests as we want those logs to debug while we have xfailed the tests.

### What is the impact of this change?
Better customer experience!

### How was this change tested?

See [DEVELOPMENT.md](https://github.com/aws-deadline/deadline-cloud/blob/mainline/DEVELOPMENT.md#testing) for information on running tests.

- Have you run the unit tests? yes
- Have you run the integration tests? yes
- Have you made changes to the `download` or `asset_sync` modules? If so, then it is highly recommended
  that you ensure that the docker-based unit tests pass. N/A

### Was this change documented?

- Are relevant docstrings in the code base updated? N/A
- Has the README.md been updated? If you modified CLI arguments, for instance. N/A

### Does this PR introduce new dependencies?

This library is designed to be integrated into third-party applications that have bespoke and customized deployment environments. Adding dependencies will increase the chance of library version conflicts and incompatabilities. Please evaluate the addition of new dependencies. See the [Dependencies](https://github.com/aws-deadline/deadline-cloud/blob/mainline/DEVELOPMENT.md#dependencies) section of DEVELOPMENT.md for more details.

*   [ ] This PR adds one or more new dependency Python packages. I acknowledge I have reviewed the considerations for adding dependencies in [DEVELOPMENT.md](https://github.com/aws-deadline/deadline-cloud/blob/mainline/DEVELOPMENT.md#dependencies).
*   [x] This PR does not add any new dependencies.

### Is this a breaking change?

No
### Does this change impact security?

No

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: test/integ/cli/test_cli_incremental_download.py [hunk 1]
```diff
@@ -217,6 +220,11 @@ def test_incremental_download_many_small_files(incremental_download_test, tmp_pa
         )
 
         job_complete = task_run_status in ["SUCCEEDED", "FAILED", "CANCELED"]
+
```
Comment (sakshie95) [inline test/integ/cli/test_cli_incremental_download.py:223]: This is not a clean method so I want to ideally shift this to using a waiter - but I'm testing the memory leak hypothesis with this timeout handling & will shift this to a waiter in a follow-up.

**Comment by sakshie95** [comment]
Hatch linting is impacted by bug https://github.com/pallets/click/issues/3065#issuecomment-3309243877

**Review by leongdl — APPROVED** [review]

**Review by jairaws — APPROVED** [review]

**Review by jairaws — COMMENTED** [review]
There is a linting error that is making some of the auto validations fail.
```
TypeError: the JSON object must be str, bytes or bytearray, not Sentinel
```

**Review by mwiebe — APPROVED** [review]
